sunpy/net/fido_factory.py

```diff
@@ -186,7 +186,8 @@ def __str__(self):
             ret = f'Results from {len(self)} Providers:\n\n'
         for block in self:
             ret += f"{len(block)} Results from the {block.client.__class__.__name__}:\n"
-            if block.client.info_url is not None:
+            if block.client.__class__.__name__ == 'VSOClient' and block.client.info_url is not None:
```

Comment:
I may be misunderstanding the best way to implement this. Looking at the code:

1. I see that the output formatting is handled by __str__ in fido_factory.py:
```
  def __str__(self):
        nprov = len(self)
        if nprov == 1:
            ret = f'Results from {len(self)} Provider:\n\n'
        else:
            ret = f'Results from {len(self)} Providers:\n\n'
        for block in self:
            ret += f"{len(block)} Results from the {block.client.__class__.__name__}:\n"
            if block.client.info_url is not None:
                ret += f'Source: {block.client.info_url}\n'
            size = block.total_size()
```


2. My current understanding is that modifying VSOClient's info_url property wouldn't work well because fido_factory.py prepends "Source:" to the entire info_url output.

Could you help me understand:
1. Is this the correct interpretation of how the output formatting works?
2. What would be the best approach to add the health status link before the "Source:" line?

Thanks

Reply:
Ah okay! Thanks for clarifying. I will look into this before pushing any other commit wildly.